doc/source/how-to/testing.rst

```diff
@@ -0,0 +1,536 @@
+
+We recommend that you follow TDD when developing your PyAnsys project, and this
+document contains examples and best practices to help you write them.
+
+Testing Framework
+-----------------
+.. raw:: html
+    
+    <div align="center">
+      <img width="50%"; src="https://docs.pytest.org/en/7.1.x/_static/pytest_logo_curves.svg">
+    </div>
+
+For consistency, PyAnsys tools and libraries should use either the `unittest
+<https://docs.python.org/3/library/unittest.html>`_ or `pytest
+<https://docs.pytest.org/>`_ frameworks for unit testing. This last framework is
+recommended unless any contrain prevents you from using it in your project.  As
+described in :ref:`Required Files`, unit tests should be placed in :ref:`The
+\`\`tests/\`\` Directory` in the library's root directory.
+
+
+Adding Testing Dependencies
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Requirements for testing dependencies should be included either in :ref:`The
+\`\`setup.py\`\` File`, :ref:`The \`\`pyproject.toml\`\` File` or in a
+``requirements_tests.txt`` file. Notice that only ``pytest`` and ``pytest-cov``
+need to be specified as third party dependencies since ``unittest`` is included
+in `The Python Standard Library <https://docs.python.org/3/library/>`_.
+
+.. tabs::
+
+    .. tab:: Flit
+
+        .. code-block:: toml
+
+            [project.optional-dependencies]
+            test = [
+                "pytest",
+                "pytest-cov",
+            ]
+
+    .. tab:: Poetry
+
+        .. code-block:: toml
+
+            [tool.poetry.group.test.dependencies]
+            pytest = "*"
+            pytest-cov = "*"
+
+    .. tab:: Setuptools
+
+        .. code-block:: python
+
+            setup(
+                name="ansys-<product>-<library>",
+                ...,
+                extras_require={
+                    "test": ["pytest", "pytest-cov"],
+                },
+            )
+
+    .. tab:: Requirements
+
+        .. code-block:: text
+
+            pytest
+            pytest-cov
+
+
+These dependencies can be installed using ``pip``:
+
+.. tabs::
+
+    .. tab:: From setup.py or pyproject.toml
+
+        .. code-block:: text
+
+            python -m pip install .[test]
+
+    .. tab:: From requirements_tests.txt
+
+        .. code-block:: text
+
+            python -m pip install -r requirements_tests.txt
+
+
+Organizing Test Files
+~~~~~~~~~~~~~~~~~~~~~
+Test files must be collected in :ref:`The \`\`tests/\`\` Directory`. To
+guarantee that tests are run against the library source code, follow a ``src/``
+layout as explained in :ref:`The \`\`src/\`\` Directory` section rather than
+having your Python library source directly within the repository root directory. 
+
+This helps you avoid testing the source of the repository and
+rather the installed package. This helps to catch errors caused by files that
+might be missed by the installer, including any C extensions or additional
+internal packages.
```

Comment:
```suggestion
This helps you to:

- Avoid testing the source of the repository rather than testing the installed package
- Catch errors caused by files that might be missed by the installer, including any C extensions or additional
  internal packages.

```